project-docs/stumptown-writers-guide.md

```diff
@@ -0,0 +1,697 @@
+
+Let's look at ./recipes/html-element.yaml:
+
+```
+related_content: /content/related_content/html.yaml
+body:
+- prose.short_description
+- meta.interactive_example?
+- prose.overview
+- prose.attributes_text?
+- meta.attributes
+- prose.styling?
+- prose.accessibility_concerns?
+- prose.*
+- meta.examples
+- meta.info_box
+- meta.specifications
+- meta.browser_compatibility
+- prose.see_also
+```
+
+This describes what must be present (and in some cases what _may_ be present) in all HTML element pages, and the order in which items must appear. Each distinct type of page in MDN - CSS property pages, Web API method pages, JS function pages - will have one of these files, although some of them will be very similar.
+
+The first item, `related_content`, points to a special YAML file that describes other pages related to this one. It's used to build a sidebar for this page (but could have other uses too). See [Related content](#related-content).
+
+The other item, `body`, describes the structure of the page content. Each property under here names a piece of the page.
+
+We said before that a docs file was split into two parts: YAML front matter and Markdown prose. Correspondingly, ingredients in the recipe's `body` have one of two prefixes:
+
+-   `meta.` ingredients are supplied by the YAML front matter
+-   `prose.` ingredients are supplied by the Markdown prose
+
+Ingredients with a `?` suffix are optional.
+
+### Meta ingredients
+
+Here are just the `meta` ingredients from the recipe:
+
+```
+- meta.interactive_example?
+- meta.attributes
+- meta.examples
+- meta.info_box
+- meta.specifications
+- meta.browser_compatibility
+```
+
+Looking back at our docs file for the `<br>` element, we should see that each `meta` item has a corresponding property in the front matter:
+
+```yaml
+---
+title: "<b>: The Bring Attention To element"
+short_title: <b>
+mdn_url: /en-US/docs/Web/HTML/Element/b
+specifications: https://html.spec.whatwg.org/multipage/semantics.html#the-b-element
+interactive_example:
+    url: https://interactive-examples.mdn.mozilla.net/pages/tabbed/b.html
+    height: html-short
+browser_compatibility: html.elements.b
+examples:
+    - examples/simple-b-example
+attributes:
+    global: /content/html/global_attributes
+recipe: html-element
+---
+
+```
+
+(well, `info_box` is missing, because we haven't figured its syntax out yet.)
+
+The syntax of each item is completely specific to this item. Some syntaxes are very simple: for example `browser_compatibility` is just the BCD query string for the item. Some, like `examples` are much more complicated.
+
+Note that the front matter also contain three items not in the recipe:
+
+```yaml
+title: "<b>: The Bring Attention To element"
+short_title: <b>
+mdn_url: /en-US/docs/Web/HTML/Element/b
+```
+
+**Note: It's arguable whether these three items should actually be in the recipe, but they are applicable to every page, so it seemed redundant.**
```

Comment:
I think we don't need these 3 in the recipes. Maybe we should see them as things you'll always need no matter what kind of content you're creating, so I think I would see them de-coupled from individual page recipes, but as "basic requirements" for any page.

Reply:
Filed as https://github.com/mdn/stumptown-content/issues/203.